src/OSPSuite.Presentation/Presenters/SimulationResidualVsTimeChartPresenter.cs

```diff
@@ -0,0 +1,179 @@
+      private const string ZERO = "Zero";
+
+      public SimulationResidualVsTimeChartPresenter(ISimulationVsObservedDataView view, ChartPresenterContext chartPresenterContext,
+         IObservedDataRepository observedDataRepository, IResidualCalculatorFactory residualCalculatorFactory)
+         : base(view, chartPresenterContext, ApplicationIcons.PredictedVsObservedAnalysis,
+            PresenterConstants.PresenterKeys.SimulationPredictedVsObservedChartPresenter)
+      {
+         _observedDataRepository = observedDataRepository;
+         _residualCalculator = residualCalculatorFactory.CreateFor(new ParameterIdentificationConfiguration());
+      }
+
+      protected override void UpdateAnalysis()
+      {
+         var simulationResidual = _residualCalculator.Calculate(_simulation.ResultsDataRepository, _simulation.OutputMappings.All);
+         _allOutputResiduals = simulationResidual.AllOutputResiduals;
+         if (!getAllAvailableObservedData().Any())
+            return;
+
+         addZeroMarkerCurveToChart();
+
+         if (ChartIsBeingCreated)
+         {
+            Chart.AxisBy(AxisTypes.Y).Caption = Captions.ParameterIdentification.Residuals;
+            Chart.AxisBy(AxisTypes.Y).Scaling = Scalings.Linear;
+         }
+
+         UpdateChartFromTemplate();
+         View.SetTotalError(simulationResidual.TotalError);
+      }
+
+      protected override void AddRunResultToChart()
+      {
+         _allOutputResiduals.GroupBy(x => x.FullOutputPath).Each(addOutputToScatter);
+      }
+
+      private void addOutputToScatter(IGrouping<string, OutputResiduals> outputMappingsByOutput)
+      {
+         var fullOutputPath = outputMappingsByOutput.Key;
+         var shouldShowInLegend = true;
+         foreach (var outputMapping in outputMappingsByOutput)
+         {
+            var dataRepository = getOrCreateScatterDataRepositoryFor(outputMapping);
+            AddResultRepositoryToEditor(dataRepository);
+            var visibleInLegend = shouldShowInLegend;
+
+            AddCurvesFor(dataRepository, (column, curve) =>
+            {
+               SelectColorForPath(fullOutputPath);
+               UpdateColorForPath(curve, fullOutputPath);
+               curve.Name = fullOutputPath;
+               curve.Description = "description";
+               curve.Symbol = Symbols.Circle;
+               curve.LineStyle = LineStyles.None;
+               curve.VisibleInLegend = visibleInLegend;
+            });
+
+            shouldShowInLegend = false;
+         }
+      }
+
+      private void addZeroMarkerCurveToChart()
+      {
+         var markerRepository = createMarkerRepository();
+         AddDataRepositoriesToEditor(new[] { markerRepository });
+         AddCurvesFor(markerRepository, (column, curve) =>
+         {
+            curve.UpdateMarkerCurve(ZERO);
+            _markerCurveId = curve.Id;
+         });
+      }
+
+      private DataRepository createMarkerRepository()
```

Comment:
That really looks a lot like the other code. This does not have to be refactored in this PR but this needs to be refactored absolutely. This is an exact copy paste of the same logic. Create a service for all of those methods